### ISSUE TITLE
Modal: remove "dialog" role

### ISSUE THREAD

**Issue by hvinh** [ng-bootstrap/ng-bootstrap#1944]
### Bug description:
Accessibility issue
As raised during user tests with NVDA users, the use of "dialog" role was blocking for NVDA users.
The same issue is observed with the ng-bootstrap modal widget.
Indeed, this role is forcing NVDA into Forms mode: when the dialog appears, its content it is not announced and the usual up/down arrows keys can not be used to read the content of the dialog.

Tested with NVDA and Firefox latest versions

### Link to minimally-working plunker that reproduces the issue:

https://ng-bootstrap.github.io/#/components/modal/examples

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: angular 4.4.6

ng-bootstrap: 1.0.0-beta.5

Bootstrap: 4.0.0-beta.2

**Comment by changLiuUNSW** [comment]
If we remove role=dialog, the modal will fail to meet  WCAG2.0 4.1.2 Name, Role, Value.
Did you raise this bug/issue in Bootstap GitHub page? I believe their Jquery modal component also has role=dialog.
Actually most components in this libraries are following Bootstrap's Accessibility implementation.

**Comment by changLiuUNSW** [comment]
I think this is a bug related to NVDA, because all the accessibility implementation of modal are following WAI-ARIA Authoring Practices 1.1
https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal

**Comment by hvinh** [comment]
Hello, actually this is related to the following issue opened to NVDA:
https://github.com/nvaccess/nvda/issues/4493
It is now fixed with the last version of NVDA 2017.4
I tested it on Firefox and NVDA no longer changes to forms mode when encountering dialog role.
Hence we can now use this role.

Kind regards
